lib/stellar_core_commander/process.rb

```diff
@@ -502,13 +502,17 @@ def record_performance_metrics(fname, txtype, accounts, txs, txrate, batchsize)
       run_data.push(m["ledger.ledger.close"]["mean_rate"])
 
       write_csv fname, METRICS_HEADER unless File.file?(fname)
-      write_csv fname, run_data
+      if METRICS_HEADER.length == run_data.length
+        write_csv fname, run_data
+      else
+        $stderr.puts "#{@name}: METRICS_HEADER and run_data have different number of columns."
```

Comment:
raise instead of logging: this would be a bug in commander and you don't want the caller to find out about this bug days/weeks later that they don't have the right data in their csv files

Reply:
Done. Need to squash some commits later, but left it as is for now so it's easy to see changes.